### ISSUE TITLE
Low throughput with large messages and high latency (25ms RTT)

### ISSUE THREAD

**Issue by SartorialOffense** [grpc/grpc-dotnet#1238]
Greetings!

I am having performance issues receiving/sending large messages (~100k-100MB) over a connection with 25 ms RTT. Transfer rate is ~2 MB/s but when serving static files of the same size on the same kestrel instance >25 MB/s (tested in Chrome with stopwatch).  When testing on localhost, 200 MB/s. 

I see the issue both with large unary operations or streaming (breaking up into chunks between 10K and 10MB). Poking about in WireShark, it looks like the large static downloads dramatically increase the TCP window to compensate for the latency during the transfer while gRPC transfers don't. I assume this is the issue. I have not been able to find any settings which change this.

Any ideas? Thanks!

### The Service
```csharp
public override Task<DataAccessResponse> TestDownload(DataAccessRequest request, ServerCallContext context)
{
    _bigchunk ??= new byte[100_000_000];

    var response = new DataAccessResponse
    {
        DataChunk = ByteString.CopyFrom(_bigchunk)
    };
    
    return Task.FromResult(response);
}
```
### Server Setup
```csharp
  services.AddGrpc(grpc =>
  {
      grpc.MaxReceiveMessageSize = 100_000_000;
  });

 webBuilder.UseKestrel(options =>
  {
      //I have used the defaults and played around with these settings - no difference
      options.Limits.Http2.InitialConnectionWindowSize = int.MaxValue;
      options.Limits.Http2.InitialStreamWindowSize = int.MaxValue;
      options.Limits.Http2.MaxFrameSize = 16_777_215;
      options.AllowSynchronousIO = true;
  });
}
```

### Dependencies
```xml
    <PackageReference Include="Google.Protobuf" Version="3.13.0" />
    <PackageReference Include="Grpc.AspNetCore.Server" Version="2.35.0" />
    <PackageReference Include="Grpc.Net.Common" Version="2.35.0" />
    <PackageReference Include="Grpc.Net.ClientFactory" Version="2.35.0" />
    <PackageReference Include="Grpc.Core.Api" Version="2.35.0" />
```
.NET 5.0

**Comment by BrennanConroy** [comment]
@JamesNK Are there client side settings for windows sizes?

**Comment by davidfowl** [comment]
Possibly this https://github.com/dotnet/runtime/issues/43086

**Comment by JamesNK** [comment]
> @JamesNK Are there client side settings for windows sizes?

Nothing in gRPC. I'm not aware of anything like that in SocketsHttpHandler.

**Comment by SartorialOffense** [comment]
So I did a test using HttpClient by itself in .NET 5.0 and on a large static file on kestrel and it matches the speed I see in gRPC (~2-3 MB/s). If I download this same file in Chrome, I see >15 MB/s. This indeed seems to be the https://github.com/dotnet/runtime/issues/43086 issue like @davidfowl suggested.

**Comment by JamesNK** [comment]
Ok, thanks for double checking. This isn't something that can be fixed in gRPC. It will need to be changed in HttpClient in dotnet/runtime.

**Comment by SartorialOffense** [comment]
Thanks for the help, I have workarounds until this is fixed in hopefully 6.0.

**Comment by davidfowl** [comment]
> BTW, googled you and realized we are both Florida Tech alums (although back in the stone ages we called it FIT).

Panthers unite!
